### ISSUE TITLE
Tariff lifecycle management

### ISSUE THREAD

**Issue by larwa99** [RI-SE/Eltariff-API#9]
I princip alla tariffer gäller under en viss tid (t.ex. årsvis) och vid något tillfälle behöver de bytas mot en ny tariffdefinition med justerade datum och/eller priser. För att förenkla logik baserat på det unika id som är kopplat till varje tariff bör vi antingen:
1. Rekommendera att tariffens 'id' behålls när man uppdaterar en viss tariff.
2. Införa stöd för att ange en koppling till det historiska 'id' som tariffen ersätter.

Jag är av åsikten att tariffens 'id' inte bör behållas (alt 1), dvs en API-konsument ska vara säker på att tariffdefinitionen inte har ändrats om 'id' inte har ändrats. Jag föredrar istället att kunna koppla ihop en ny tariff med en redan existerande (alt 2) för att enklare kunna migrera över till den nya genom att t.ex. uppdatera en kunddatabas med det nya värdet när det börjar gälla.
Förslag på optional property 'supersedes' enligt nedan:

````
"tariffs": [
  {
    "id": "4891692b-46ec-4074-8a08-e0288b363a0a",
    "name": "Prislista 16A standard",
    "supersedes": "111530c9-a274-4f97-97ad-978d5345da27",
 ...
]
````

**Comment by larwa99** [comment]
I have looked into this and a preferred alternative to 'supersedes' is to have a persistent reference that allows an API consumer to have traceability of new versions for the same tariff/product. My suggestion is to name it 'productReference' instead of 'reference' which is used for other datastructures in the API since it has different requirements on being unique and persistent across tariff releases.

Definition to be added to [tariff.schema.json](https://github.com/RI-SE/Eltariff-API/blob/main/specification/tariff.schema.json):
```
        "productReference": {
          "description": "A persistent and unique (on company level) reference to identify a given product during its entire lifecycle. If set, the reference allows for a seamless transition when a product's tariff definition gets updated.",
          "type": "string",
          "examples": [
            "net.lkpg.cons.std.25",
            "lkpg-cons-hsp-max10000",
            "b960a908-e7dc-4032-8ffd-1e56eacc9916"
          ],
        }
```

Example of use:
```
"tariffs": [
  {
    "id": "4891692b-46ec-4074-8a08-e0288b363a0a",
    "name": "Prislista 16A standard",
    "productReference": "net.lkpg.cons.std.16",
 ...
]
```

FYI: @portcallbuilder @mattiaspalmqvist

**Comment by larwa99** [comment]
@mattiaspalmqvist An alternative to introduce the new `productReference` is to change the definition of the existing `product` so it is clearly intended to be used as a unique and persistent reference. What's your thought on this, is it doable based on how you currently use it in Göteborg Energi?

**Comment by mattiaspalmqvist** [comment]
Yes, that will work for us. I think it's a good idea to use the product property. I have changed the description of the product property in the API specification to be used as this unique identifier.

Check if it looks good, then this issue can be closed.

**Comment by larwa99** [comment]
It looks very good, let's use this.
